components/remote_settings/src/client.rs

```diff
@@ -20,6 +21,28 @@ const HEADER_BACKOFF: &str = "Backoff";
 const HEADER_ETAG: &str = "ETag";
 const HEADER_RETRY_AFTER: &str = "Retry-After";
 
+#[derive(Debug, Clone, Deserialize)]
+pub struct PackagedData {
+    collections: HashMap<String, CollectionData>,
+}
+
+impl PackagedData {
+    pub fn from_json_str(json: &str) -> Result<Self> {
+        Ok(serde_json::from_str(json)?)
+    }
+
+    pub fn empty() -> Self {
+        Self {
+            collections: HashMap::new(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct CollectionData {
+    records: Vec<RemoteSettingsRecord>,
+}
```

Comment:
Could we also load the `last_modified` timestamp?  I think this would simplify checking if this is newer or older than the cached records.